python/tvm/relay/frontend/tflite.py

```diff
@@ -0,0 +1,672 @@
+        filter_h = pool2d_options.FilterHeight()
+        filter_w = pool2d_options.FilterWidth()
+        fused_activation_fn = pool2d_options.FusedActivationFunction()
+
+        params = {'pool_size': (filter_h, filter_w),
+                  'strides': (stride_h, stride_w),
+                  'padding': [0, 0]}
+
+        in_expr = self.get_expr(input_tensor_idx)
+
+        _, input_h, input_w, _ = input_tensor.tensor.ShapeAsNumpy()
+        if padding == Padding.VALID:
+            pass
+        elif padding == Padding.SAME:
+            pad_top, pad_bottom = get_pad_value(input_h, filter_h, stride_h)
+            pad_left, pad_right = get_pad_value(input_w, filter_w, stride_w)
+            params['padding'] = [pad_top, pad_left, pad_bottom, pad_right]
+        else:
+            raise NotImplementedError("Not support padding format: {}".format(padding))
+
+        if pool_type == "average":
+            out = _op.nn.avg_pool2d(in_expr, **params)
+        elif pool_type == "max":
+            out = _op.nn.max_pool2d(in_expr, **params)
+        else:
+            raise ValueError("Not support pool type: {}".format(pool_type))
+
+        # If we have fused activations
+        if fused_activation_fn != ActivationFunctionType.NONE:
+            out = self.convert_fused_activation_function(out, fused_activation_fn)
+
+        return out
+
+    def get_expr(self, input_tensor_idx):
+        return self.exp_tab.get_expr(get_tensor_name(self.subgraph, input_tensor_idx))
+
+def build_str_map(obj):
+    """Build string map of TFLite enum int value
+
+    Parameters
+    ----------
+    obj:
+        TFLite class which contains enum int value, such as BuiltInOptions
+
+    Returns
+    -------
+        String representation map of TFLite class enum int value
+    """
+    ret = {}
+    for field_name in dir(obj):
+        if not field_name.startswith('_'):
+            field_value = getattr(obj, field_name)
+            if isinstance(field_value, int):
+                ret[field_value] = field_name
+    return ret
+
+# SAME padding: https://www.tensorflow.org/api_guides/python/nn
+def get_pad_value(data, kernel, stride):
+    """Get the pad tuple of value for SAME padding
+
+    Parameters
+    ----------
+    data:
+        1D input data
+
+    kernel:
+        1D input kernel
+
+    stride:
+        1D input stride
+
+    Returns
+    -------
+        pad tuple of value
+    """
+
+    out = math.ceil(float(data) / float(stride))
+    pad = max(0, (out - 1) * stride + kernel - data)
+    pad_before = pad // 2
+    pad_after = pad - pad_before
+    return pad_before, pad_after
+
+
+def get_tensor_name(subgraph, tensor_idx):
+    """Get the tensor name.
+
+    Parameters
+    ----------
+    subgraph:
+        tflite.Subgraph.Subgraph
+
+    tensor:
+        tensor index in subgraph
+
+    Returns
+    -------
+        tensor name in UTF-8 encoding
+    """
+    return subgraph.Tensors(tensor_idx).Name().decode("utf-8")
```

Comment:
@FrozenGene I am a little bit surprised that this could pass python3 CI, because `str.decode` shouldn't exist in Python3...Any ideas? Is the resulted type of `Name()` `str` or `bytearray` in Python3?

Reply:
The resulted type of `Name()` is not `str`, it is `bytes()` in TFLite. See: https://github.com/FrozenGene/tflite/blob/master/tflite/Tensor.py#L58

Reply:
Gotcha. So it is reasonable to me. Thanks!